internal/cmd/commands/database/funcs.go

```diff
@@ -43,6 +43,7 @@ func migrateDatabase(ctx context.Context, ui cli.Ui, dialect, u string, initiali
 		}
 		return noop, 2
 	}
+	defer man.Close(ctx)
```

Comment:
Hmm, while it probably would be good to call this `Close`, I think this might actually cause a new issue. A few lines down we define `unlock` function, which will try to use `man` to unlock the exclusive lock. This function is returned and then called by the callter of `migrateDatabase`. If we do this `man.Close`, the connection will be closed prior to the attempt to call `man.ExclusiveUnlock`.

So we need to do a bit more here. Perhaps we could include the call to `man.Close` in the `unlock` function. And we would still need to correctly call close in the error case below if we are unable to get an ExclusiveLock.

Reply:
Ah, good thinking. I'll just close this manually in each case.